bellows/zigbee/application.py

```diff
@@ -277,4 +279,24 @@ def permit_with_key(self, node, code, time_s=60):
         if v[0] != t.EmberStatus.SUCCESS:
             raise Exception("Failed to change policy to allow generation of new trust center keys")
 
+        yield from self.send_zdo_broadcast(0x0036, 0x0000, 0x00, [time_s, 0])
         return self._ezsp.permitJoining(time_s, True)
+
+    @asyncio.coroutine
+    def send_zdo_broadcast(self, command, grpid, radius, args):
+        aps_frame = t.EmberApsFrame()
+        aps_frame.profileId = t.uint16_t(0x0000)       # 0 for zdo
```

Comment:
Is there any reason that logic like this can't be encapsulated in the ZDO layer? I guess the whole device-oriented API doesn't make it natural to do that, but it still seems to be that adding ZDO-specific stuff into the application layer will make it a little awkward to deal with other kinds of broadcasts later.

I'd certainly be happy to have a `send_broadcast` inside the application. Similar to "request", I think it should be agnostic of whether it's ZCL or ZDO (or anything else that can be dreamed up at that layer).

Reply:
Ok, I think this makes sense. let me create a more generic send_broadcast (need this anyway for some multicast/group stuff I'm working on) and create a send_broadcast inside the ZDO section.  
I will keep EZSP specific stuff (APS frame, and ezsp_sendmulticast ) inside bellows and the other functions inside zigpy. 
Ok, stay tuned